[1.19.x] Document Forge extensions of the model JSON format (#493)
MinecraftForge/Documentation: docs/rendering/modelextensions/index.md

Model Extensions
================

> Forge provides several extensions to the model JSON format to allow fine tuning the way JSON models are loaded, baked and rendered:

Reviewer: ```suggestion
Forge provides several extensions to vanilla's JSON model format to allow fine tuning of loading, baking and rendering behavior:
```

- [Custom Model Loaders][modelloaders]
- [Root Transforms][transforms]
- [Render Types][rendertypes]
- [Part Visibility][visibility]
- [Face Data][facedata]

[modelloaders]: ../modelloaders/index.md
[transforms]: ./transforms.md
[rendertypes]: ./rendertypes.md
[visibility]: ./visibility.md

> [facedata]: ./facedata.md

Reviewer 2: This introduction section doesn't feel as though it is introducing anything. This should either hold the docs for something, like vanilla extensions, or just shouldn't exist at all since it can be seen through the sidebar.

Author: I'm not satisfied with the introduction section because, yes, it's basically empty. But for some reason it felt weird to not have an introduction with a short overview. I'll have to find a way to meaningfully expand on this.
